### PR TITLE
Adam9500/salus 606 or label match

### PR DESCRIPTION
**Pull request by harageth**
# Resolves

SALUS-606

# What

Adds the logical OR operator to agent catalog management so that we can replicate the same results in this service as what will happen in resource management for both logical operators.

# How

It adds the query here and does the same logic the monitor management is doing

## How to test

Run the unit tests. 

# Why

This was the same way we handled this in other services

### CODE REVIEW

File: src/main/java/com/rackspace/salus/acm/services/AgentInstallService.java [hunk 1]
```diff
@@ -166,17 +169,17 @@ void handleResourceEvent(ResourceEvent resourceEvent) {
     }
   }
 
-  List<AgentInstall> getInstallsFromLabels(String tenantId, Map<String, String> labels)
+  List<AgentInstall> getInstallsFromResourceLabels(String tenantId, Map<String, String> resourceLabels)
       throws IllegalArgumentException {
-    if (labels.size() == 0) {
+    if (resourceLabels.size() == 0) {
```
Comment (harageth) [inline src/main/java/com/rackspace/salus/acm/services/AgentInstallService.java]: Did we need to do more validation here?

File: src/main/resources/sql-queries/agent_installs_label_matching_OR_query.sql [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+               tenant_id = :tenantId
+         )
+         AND agent_installs.id IN
+         (
+            SELECT
+               search_labels.agent_install_id
+            FROM
+                (
+                   SELECT
+                      agent_install_id,
```
Comment (harageth) [inline src/main/resources/sql-queries/agent_installs_label_matching_OR_query.sql]: We should be explicit in these column declarations like we are with the other ones.

File: src/test/java/com/rackspace/salus/acm/services/AgentReleaseServiceTest.java [hunk 3]
```diff
@@ -157,6 +158,7 @@ private AgentInstall saveInstall(AgentRelease release, String tenantId,
             .setAgentRelease(release)
             .setTenantId(tenantId)
             .setLabelSelector(labelSelector)
+            .setLabelSelectorMethod(LabelSelectorMethod.AND)
```
Comment (harageth) [inline src/test/java/com/rackspace/salus/acm/services/AgentReleaseServiceTest.java]: Use the passed in value here.

File: src/test/java/com/rackspace/salus/acm/web/controller/AgentInstallControllerTest.java [hunk 4]
```diff
@@ -188,6 +190,7 @@ private AgentInstall populateInstall(AgentRelease release) {
     return new AgentInstall()
         .setId(UUID.fromString("00000000-0000-0000-0002-000000000000"))
         .setLabelSelector(singletonMap("os", "linux"))
+        .setLabelSelectorMethod(LabelSelectorMethod.AND)
```
Comment (harageth) [inline src/test/java/com/rackspace/salus/acm/web/controller/AgentInstallControllerTest.java]: We should probably add a test for the OR case here.

File: cloudbuild.yaml [hunk 5]
```diff
@@ -17,38 +17,39 @@ steps:
...
+          tar -xzf /tmp/m2-cache.tar.gz
+        ) || echo 'Cache not found'
     volumes:
-    - name: user.home
-      path: /root
+      - name: user.home
+        path: /root
 
-  - id: DEPLOY
+  - id: VERIFY
```
Comment (harageth) [inline cloudbuild.yaml]: I should also ask why this change was made as it doesn't seem to have relevance to the rest of the PR

File: src/main/java/com/rackspace/salus/acm/services/AgentInstallService.java [hunk 6]
```diff
@@ -197,6 +200,14 @@ void handleResourceEvent(ResourceEvent resourceEvent) {
         (resultSet, rowIndex) -> UUID.fromString(resultSet.getString(1))
     );
 
+    final List<UUID> monitorOrIds = namedParameterTemplate.query(String.format(labelMatchORQuery, builder.toString()), paramSource,
+        (resultSet, rowIndex) -> UUID.fromString(resultSet.getString(1))
+    );
+
+    monitorIds.addAll(monitorOrIds);
+
+    //combine the lists of UUID's here
```
Comment (GeorgeJahad) [inline src/main/java/com/rackspace/salus/acm/services/AgentInstallService.java]: is this comment referring to the line 207?

Comment (harageth) [reply src/main/java/com/rackspace/salus/acm/services/AgentInstallService.java]: Yes it is. That should get moved.

File: src/main/resources/sql-queries/agent_installs_label_matching_OR_query.sql [hunk 7]
```diff
@@ -0,0 +1,31 @@
+SELECT agent_installs.id
+FROM   agent_installs JOIN agent_install_label_selectors AS ail
+WHERE  agent_installs.id = ail.agent_install_id
+AND agent_installs.label_selector_method = 'OR'
+AND agent_installs.id IN
+(
+  SELECT ails.agent_install_id
```
Comment (itzg) [inline src/main/resources/sql-queries/agent_installs_label_matching_OR_query.sql]: Indentation is much more readable on this one. Thanks.

File: src/main/resources/sql-queries/agent_installs_label_matching_OR_query.sql [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    FROM   agent_installs AS ai
+    WHERE  ai.tenant_id = :tenantId
+  )
+  AND    agent_installs.id IN
+  (
+    SELECT   search_labels.agent_install_id
+    FROM
+      (
+        SELECT  inner_ails.agent_install_id,
+          COUNT(*) AS count
```
Comment (GeorgeJahad) [inline src/main/resources/sql-queries/agent_installs_label_matching_OR_query.sql]: I'm curious about the differences between this clause and the corresponding clauses in the monitor version.  should they be the same?

```
AND    agent_installs.id IN
  (
    SELECT   search_labels.agent_install_id
    FROM
      (
        SELECT  inner_ails.agent_install_id,
          COUNT(*) AS count
        FROM     agent_install_label_selectors AS inner_ails
        WHERE    %s
        GROUP BY inner_ails.agent_install_id
      )
      AS     search_labels
    WHERE    search_labels.count >= 1
    GROUP BY search_labels.agent_install_id
  )
)```

Comment (harageth) [reply src/main/resources/sql-queries/agent_installs_label_matching_OR_query.sql]: There is no difference. Not sure how I let this slip by.
